include/picongpu/particles/collision/binary/DefaultAlg.hpp

```diff
@@ -0,0 +1,494 @@
+/* Copyright 2013-2019 Rene Widera, Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/particles/collision/binary/DefaultAlg.def"
+
+#include <pmacc/random/distributions/Uniform.hpp>
+
+#include <utility>
+#include <type_traits>
+#include <cmath>
+#include <stdio.h>
+#include <fenv.h>
+
+namespace picongpu
+{
+namespace particles
+{
+namespace collision
+{
+namespace binary
+{
+namespace acc
+{
+    using namespace pmacc;
+    struct DefaultAlg
```

Comment:
This functor and file should not be called `DefaultAlg`, please give it a name that reflects the method used.